components/Chat.tsx

```diff
@@ -63,34 +71,33 @@ export default function Chat() {
         throw new Error(`HTTP error! status: ${res.status}`);
       }
 
-      const { id } = await res.json()
+      const { id } = await res.json();
 
       const interval = window.setInterval(async () => {
```

Comment:
Lets add a try/catch within this interval which will clean up the set interval in the event of an error

Reply:
Then we can move the duplicated cleanup code into a finally block

Reply:
I found it cleaner to throw an exception that propagates to the handleSubmit's try/catch, so we can use its `catch` clause. We can't use finally block to de-duplicate the cleanup code because when there's no error, we need the interval to continue.